pkg/config/trustengine.go

```diff
@@ -0,0 +1,52 @@
+package config
+
+import (
+	"errors"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/trustengine"
+)
+
+// resolveAllTrustEngineReferences retrieves all the step's secrets from the Trust Engine
+func resolveAllTrustEngineReferences(config *StepConfig, params []StepParameters, trustEngineConfiguration trustengine.Configuration, client *piperhttp.Client) {
+	for _, param := range params {
+		if ref := param.GetReference(trustengine.RefTypeSecret); ref != nil {
+			if config.Config[param.Name] == "" { // what if Jenkins set the secret?
```

Comment:
Check and remove this comment.

Reply:
Jenkins should set the secrets as env var which are picked up by cobra and passed in as parameter inputs. So no issue in my eyes.